src/main/java/com/mgmtp/radio/controller/v1/StationController.java

```diff
@@ -63,7 +63,7 @@ public StationController(StationService stationService, UserMapper userMapper, C
     @GetMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
     public Mono<StationDTO> getStation(@PathVariable(value = "id") String stationId) throws RadioNotFoundException {
-        return this.stationService.findById(stationId);
+        return this.stationService.findById(stationId).switchIfEmpty(Mono.error(new RadioNotFoundException("Station id not found!")));
```

Comment:
Please change it to something that familiar to end-user, because we will use it at frontend, such as: "Station not found".